Midproject

claudia1462656918 opened the pull request:
src/get_data.py: download data from public S3 bucket
* file_key: bank.csv
* bucket_name: value-chain-project-data
* output_file_path: output path for the downloaded data ( ../data)


src/upload_data.py: upload data to your own S3 bucket
* input_file_path: local path for uploaded data
* bucket_name: the S3 bucket name that we want to put the data to
* output_file_path: output path for uploaded file on S3

src/sql/models.py: create database
* RDS True if you want to create database in RDS else None.

** Created database can be checked in 'src/sql/logfile'

Commit:
midpoint

Commit:
midpoint_check

claudia1462656918 committed:
Update database_schema.py

correct a type in variable

claudia1462656918 committed:
Add files via upload

Add the deck to the midproject

claudia1462656918 committed:
upload data

zionward commented:
When I tried to run src/get_data.py, I meet NoCredentialsError. Please check your S3 setting for download data path.
src/upload_data.py has indentation in line 10, so the file need to be modified.
src/sql/models.py does not exist, but I found src/sql/database_schema.py in src/sql. When I run this file, I got ValueError (invalid literal for int() with base 10: 'None')  for line 131, in <module>
    create_db(args), so the file need to be modified.

Commit:
modified

claudia1462656918 committed:
Update requirements.txt

claudia1462656918 committed:
Update requirements.txt

add more

claudia1462656918 committed:
Create README.md

Add the file to explain to someone else how to can get the code, build it, and run it

claudia1462656918 committed:
Update README.md

claudia1462656918 committed:
Update README.md

claudia1462656918 committed:
Update README.md

claudia1462656918 committed:
Update README.md

Add README.md

claudia1462656918 committed:
Update README.md

Modified

zionward commented:
Currently, all three file work properly and the outputs look as expected. The code is clear and easy to follow: different functionalities are separated in different functions and the names of the functions and files are meaningful and clear what they are referencing, which is very user-friendly.

The structure of the project is clear, so I can understand the design decisions that were made in the code base as a whole. Also, the code makes sense in the large context of the application. Each part are useful for software development. I do not see any issues that will impact future development now.

Currently functionalities are straightforward so testing is not necessary. For modeling part, it might be more helpful to include testing part in this project.

The documentation looks good overall, but I would suggest to add more documentation in future codes to make them easier to follow. With Docstring and comments, readers can understand the functionalities quicker and do not need to read the code line by line. For example, the docstring in create_db of database_schema.py looks clear and in proper format, so I can understand the function easily. In this stage, get_data.py and upload_data.py are short so it does not matter to read without docstring, but for future more complicated code, I would recommend she to add add more documentation. The inputs and outputs aren explained and referenced by the parameter name, which makes codes easier to follow. Besides, please add the README of this PR to README of the whole project, so that the users can run the project easily. The commit information is short but meaningful, I can know the timeline and updating from them.

claudia1462656918 committed:
Update README.md
